File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task169_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.*;

public class Solution {
    public List<Integer> countSmaller(int[] nums) {
        Integer[] indices = new Integer[nums.length];
        for (int i = 0; i < nums.length; i++) {
            indices[i] = i;
        }
        Arrays.sort(indices, (a, b) -> nums[b] - nums[a]);
        int[] counts = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            counts[indices[i]] = i;
        }
        return Arrays.asList(counts);
    }
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task169_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
class Solution {
public:
    vector<int> countSmaller(vector<int>& nums) {
        vector<pair<int, int>> enumVec;
        for (int i = 0; i < nums.size(); i++) {
            enumVec.push_back({nums[i], i});
        }
        sort(enumVec.begin(), enumVec.end(), greater<pair<int, int>>());
        vector<int> counts(nums.size(), 0);
        for (int i = 0; i < nums.size(); i++) {
            counts[enumVec[i].second] = i;
        }
        return counts;
    }
};